### PR TITLE
Feature/fr remove legacy action conversion

### PR DESCRIPTION
**Pull request by shahanemahesh**

### CODE REVIEW

File: Ginger/Ginger/SolutionWindows/TreeViewItems/BusinessFlowTreeItem.cs [hunk 1]
```diff
@@ -139,7 +141,36 @@ private void ActionsConversionHandler(object sender, System.Windows.RoutedEventA
...
+
+        /// <summary>
+        /// This method helps to execute the funcationality of removeing the legacy actions from the businessflow
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void LegacyActionsRemoveHandler(object sender, System.Windows.RoutedEventArgs e)
+        {
+            ObservableList<BusinessFlowToConvert> lstBFToConvert = new ObservableList<BusinessFlowToConvert>();
+            if (((ITreeViewItem)this).NodeObject().GetType().Equals(typeof(GingerCore.BusinessFlow)))
```
Comment (MeniKadosh1) [inline Ginger/Ginger/SolutionWindows/TreeViewItems/BusinessFlowTreeItem.cs]: 1- why you have this IF-Else? code in IF should be here (BusinessFlowTreeItem) and else code should be in BusinessFlowFolderTreeITem, they can't be both so no need IF-Else and duplicated code

2- you dont need to get the tree item object like ((ITreeViewItem)this).NodeObject(), you have "mBusinessFlow", and "mBusinessFlowFolder" (in folderTreeItem) for that

File: Ginger/Ginger/SolutionWindows/TreeViewItems/BusinessFlowTreeItem.cs [hunk 2]
```diff
@@ -108,6 +109,7 @@ void ITreeViewItem.SetTools(ITreeView TV)
                 AddItemNodeBasicManipulationsOptions(mContextMenu);
                 MenuItem actConversionMenu = TreeViewUtils.CreateSubMenu(mContextMenu, "Conversion");
                 TreeViewUtils.AddSubMenuItem(actConversionMenu, "Legacy Actions", ActionsConversionHandler, null, eImageType.Convert);
+                TreeViewUtils.AddSubMenuItem(actConversionMenu, "Remove Legacy Actions", LegacyActionsRemoveHandler, null, eImageType.Convert);
```
[the remark this answers is not in the document]
Comment (shahanemahesh) [reply Ginger/Ginger/SolutionWindows/TreeViewItems/BusinessFlowTreeItem.cs]: Done

File: Ginger/Ginger/SolutionWindows/TreeViewItems/BusinessFlowsFolderTreeItem.cs [hunk 3]
```diff
@@ -143,6 +144,7 @@ void ITreeViewItem.SetTools(ITreeView TV)
 
                 MenuItem actConversionMenu = TreeViewUtils.CreateSubMenu(mContextMenu, "Conversion");
                 TreeViewUtils.AddSubMenuItem(actConversionMenu, "Legacy Actions", ActionsConversionHandler, null, eImageType.Convert);
+                TreeViewUtils.AddSubMenuItem(actConversionMenu, "Remove Legacy Actions", LegacyActionsRemoveHandler, null, eImageType.Convert);
```
[the remark this answers is not in the document]
Comment (shahanemahesh) [reply Ginger/Ginger/SolutionWindows/TreeViewItems/BusinessFlowsFolderTreeItem.cs]: Done

File: Ginger/Ginger/SolutionWindows/TreeViewItems/BusinessFlowTreeItem.cs (hunk 1 above)
Comment (shahanemahesh) [reply Ginger/Ginger/SolutionWindows/TreeViewItems/BusinessFlowTreeItem.cs]: Changed

**Pull request merged**